pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/rest/api/SourcesImpl.java

```diff
@@ -728,20 +729,28 @@ public SourceConfig getSourceInfo(final String tenant,
             }
         }
 
-        // if source is not builtin, attempt to extract classloader from package file if it exists
-        if (classLoader == null && sourcePackageFile != null) {
-            classLoader = getClassLoaderFromPackage(sourceConfig.getClassName(),
-                    sourcePackageFile, worker().getWorkerConfig().getNarExtractionDirectory());
-        }
+        boolean shouldCloseClassLoader = false;
+        try {
+            // if source is not builtin, attempt to extract classloader from package file if it exists
+            shouldCloseClassLoader = true;
```

Comment:
here the `ClassLoader` must be closed only if `classLoader` is null, otherwise you will close the `ClassLoader` of the `Connector` instance

Reply:
thanks for catching that @nicoloboschi . I pushed a fix.